homeassistant/components/emulated_hue/upnp.py

```diff
@@ -60,13 +68,26 @@ class UPNPResponderThread(threading.Thread):
 
     _interrupted = False
 
-    def __init__(self, host_ip_addr, listen_port, upnp_bind_multicast):
+    def __init__(self, host_ip_addr, listen_port, upnp_bind_multicast,
+                 advertise_ip, advertise_port):
         """Initialize the class."""
         threading.Thread.__init__(self)
 
         self.host_ip_addr = host_ip_addr
         self.listen_port = listen_port
         self.upnp_bind_multicast = upnp_bind_multicast
+        self.advertise_ip = advertise_ip
+        self.advertise_port = advertise_port
+
+        # If using the advertisement overides, only
+        # use them for forming the responses
+        resp_ip = self.advertise_ip if \
```

Comment:
This logic can be simplified to `resp_ip = self.advertise_ip or self.host_ip_addr`.

Also, if we move this logic into the [component setup](https://github.com/jeremydk/home-assistant/blob/19698293ee47e790d3ffa75169de33bbbc20bca4/homeassistant/components/emulated_hue/__init__.py#L91), we can avoid duplicating it by passing the response ip and response port into the `DescriptionXmlView` constructor.